service-common/src/app.rs

```diff
@@ -25,6 +26,31 @@ macro_rules! main {
 Drogue IoT {} - {} {} ({})
 "#, drogue_cloud_service_api::version::VERSION, NAME, VERSION, DESCRIPTION);
 
+        $crate::app::init_tracing(NAME);
+
         return $run;
     }};
 }
+
+#[cfg(feature = "jaeger")]
+pub fn init_tracing(name: &str) {
+    use crate::tracing::tracing_subscriber::prelude::*;
+
+    log::info!("Enabling Jaeger tracing");
+    crate::tracing::opentelemetry::global::set_text_map_propagator(
+        crate::tracing::opentelemetry_jaeger::Propagator::new(),
+    );
+    let tracer = crate::tracing::opentelemetry_jaeger::new_pipeline()
+        .with_service_name(name)
+        .install_batch(crate::tracing::opentelemetry::runtime::Tokio)
```

Comment:
with_trace_config() would be needed here to help with the sampling config 

https://docs.rs/opentelemetry-jaeger/latest/opentelemetry_jaeger/struct.PipelineBuilder.html#method.with_trace_config
https://docs.rs/opentelemetry/0.17.0/opentelemetry/sdk/trace/struct.Config.html

Reply:
I assumed this would get configured from the env-vars: https://github.com/open-telemetry/opentelemetry-rust/tree/main/opentelemetry-jaeger#jaeger-exporter-from-environment-variables

Reply:
Ah indeed : OTEL_TRACES_SAMPLER

Reply:
Not sure it works though :)